### ISSUE TITLE
Implement additional exercises

### ISSUE THREAD

**Issue by BNAndras** [exercism/racket#297]
CLAIMED
- [ ] flatten-array - BNAndras
- [ ] queen-attack - dreig

COMPLETED
- [x] bank-account
- [x] binary-search 
- [x] clock
- [x] diamond
- [x] food-chain
- [x] high-scores
- [x] kindergarten-garden
- [x] knapsack
- [x] luhn
- [x] minesweeper
- [x] pangram
- [x] parallel-letter-frequency
- [x] pascals-triangle
- [x] pig-latin
- [x] protein-translation
- [x] proverb
- [x] robot-simulator
- [x] run-length-encoding
- [x] secret-handshake
- [x] sieve
- [x] spiral-matrix
- [x] square-root
- [x] strain
- [x] two-bucket
- [x] yacht
- [x] zebra-puzzle

 Don't auto-close the issue please.

**Comment by Adrien-LUDWIG** [comment]
I only know Racket through the #12in23 challenge and did 10 exercises... But I would be happy to try implementing one! 😄
I think I could try to implement binary-search, for example.

This would be the first time I create an exercise. I already read most of the documentation. 

I could not make the "generate" command of the configlet work, but I successfully used `sync` and `uuid` to create `.docs/instructions.md`, `.docs/introduction.md`, `.meta/config.json` and `.meta/tests.toml`.
If I'm not mistaken, I need to add `binary-search.rkt` (the slug), `binary-search-test.rkt` and `.meta/example.rkt`.

**Comment by BNAndras** [comment]
Yeah, that's correct. There's no test generator so you'll need to create those files by hand. You can use another practice exercise to get an idea of the general structure for those files. Of course, I'll be glad to review anything even a work in progress. You can ping me here or on the Exercism Discord (same username).

I'll claim robot-simulator for next week. I'm holding off on bank account since I need to work on the concurrency aspect.

**Comment by Adrien-LUDWIG** [comment]
Oh, now I see why I misunderstood `generate`. [The documentation about Generating Documents](https://exercism.org/docs/building/configlet/generating-documents#:~:text=introduction.md%20file-,When%20configlet%20detects%20that%20a%20Practice%20Exercise%20implements%20a%20Problem%20Specifications,file.,-Document%3A%20Practice%20Exercise%27s) talks about exercises' from Problem Specifications having an `introduction.md`. But now I see they are synced, not generated.

I'll try to create a WIP PR soon!

**Comment by Adrien-LUDWIG** [comment]
I'm now working on `proverb`. 😉

**Comment by BNAndras** [comment]
Sounds great. I missed that `protein-translation` isn't implemented so that might be an easy one too. The Clojure and Common Lisp tracks already ported it so one could start from their examples and Racket-ify them. The Emacs Lisp track doesn't have that exercise implemented so that'd be a great follow-up PR.

**Comment by BNAndras** [comment]
But you're welcome to implement any other exercises I haven't listed. This issue simply started off as my "wouldn't these be nice to add" list.

**Comment by Adrien-LUDWIG** [comment]
I'll take a look at `protein-translation` next then!

I might not continue at this pace after Hacktoberfest (since I have to find a job too :sweat_smile:) but I'll try to keep adding one from time to time. It helped me take the plunge! 😄

**Comment by BNAndras** [comment]
Picking this back up. I'll work through flatten-array, robot-simulator, and square-root next week.

**Comment by dreig** [comment]
Can I contribute here too? I'd start for `queen-attack` if that's OK?

**Comment by BNAndras** [comment]
Yeah, go ahead. :)
